src/data_chain/chain.rs

```diff
@@ -185,3 +171,309 @@ impl Chain {
+            )?;
+        }
+        writeln!(formatter, "}}")
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use super::super::tests;
+    use data_chain::{NodeState, Proof, State};
+    use full_info::FullInfo;
+    use maidsafe_utilities::SeededRng;
+    use rand::Rng;
+    use routing_table::Prefix;
+    use rust_sodium;
+
+    #[ignore]
+    #[test]
+    fn pending_blocks() {
+        // Start with the chain having Live(A).  Create blocks for Offline(A), Live(B), and
+        // Relocated(B), Live(C).  Add the four of them in reverse order and assert after each is
+        // added that they're held in the `pending_blocks` until the final one is added and they can
+        // all be added to the chain in the order defined above.
+        let mut rng = SeededRng::thread_rng();
+        unwrap!(rust_sodium::init_with_rng(&mut rng));
+
+        let prefix = Prefix::default().pushed(rng.gen()).with_version(rng.gen());
+        let nodes = tests::create_full_infos(Some(prefix));
+
+        // Create Live(A) and Offline(A).
+        let node_a = FullInfo::within_range(5, &prefix.lower_bound(), &prefix.upper_bound());
+        let mut payload = NodeState {
+            state: State::ElderLive,
+            public_key: *node_a.public_info().sign_key(),
+            age: node_a.public_info().age(),
+            section: prefix,
+        };
+        let live_a = tests::create_block(payload, &nodes, false);
+        payload = NodeState {
+            state: State::ElderOffline,
+            public_key: *node_a.public_info().sign_key(),
+            age: node_a.public_info().age(),
+            section: prefix,
+        };
+        let offline_a = tests::create_block(payload, &nodes, false);
+
+        // Create Live(B) and Relocated(B).
+        let node_b = FullInfo::within_range(6, &prefix.lower_bound(), &prefix.upper_bound());
+        payload = NodeState {
+            state: State::ElderLive,
+            public_key: *node_b.public_info().sign_key(),
+            age: node_b.public_info().age(),
+            section: prefix,
+        };
+        let live_b = tests::create_block(payload, &nodes, false);
+        payload = NodeState {
+            state: State::ElderRelocated,
+            public_key: *node_b.public_info().sign_key(),
+            age: node_b.public_info().age(),
+            section: prefix,
+        };
+        let relocated_b = tests::create_block(payload, &nodes, false);
+
+        // Create Live(C).
+        let node_c = FullInfo::within_range(7, &prefix.lower_bound(), &prefix.upper_bound());
+        payload = NodeState {
+            state: State::ElderLive,
+            public_key: *node_c.public_info().sign_key(),
+            age: node_c.public_info().age(),
+            section: prefix,
+        };
+        let live_c = tests::create_block(payload, &nodes, false);
+
+        // Create chain containing just Live(A).
+        let mut chain = Chain::from_blocks(vec![live_a.clone()], vec![], nodes.len());
```

Comment:
not sure, but I will prefer to use `GROUP_SIZE` instead of `nodes.len()` .
same to couple of other places.

Reply:
I was thinking that if we wanted to change the group size value used in tests though, it'd be less code to change doing it this way.

Reply:
isn't change the group size value means only need to change the definition of `GROUP_SIZE` ?

Reply:
No, I mean that we can leave the default value `GROUP_SIZE` unchanged in the code, but replace its use in tests with something else more easily.

Reply:
Ah, I see.
Are those places currently passing nodes.len() actually wants `GROUP_SIZE`?
will it make more sense to use `GROUP_SIZE` as much as possible, and only use nodes.len() when really need it?
Or, if we want to have a different configuration of `GROUP_SIZE` for tests, we can define `TEST_GROUP_SIZE` (or `[cfg(test)]`) ?

just want to eliminate the confusion that passing nodes.len() gives the impression that the parameter can be any value (less/greater than GROUP_SIZE).

Reply:
We already have a dev config option for this.  How about if I do:
```rust
let group_size = nodes.len();
let mut chain = Chain::from_blocks(vec![live_a.clone()], vec![], group_size);
```